extensions/sql-database-projects/src/dialogs/publishDatabaseQuickpick.ts

```diff
@@ -159,9 +159,7 @@ export async function getPublishDatabaseSettings(project: ISqlProject, promptFor
 	}
 
 	// 4. Modify sqlcmd vars
-	// If a publish profile is provided then the values from there will overwrite the ones in the
-	// project file (if they exist)
-	let sqlCmdVariables = Object.assign({}, project.sqlCmdVariables, publishProfile?.sqlCmdVariables);
```

Comment:
after `project.sqlCmdVariables` was switched to be a Map instead of Record, this was always setting `sqlCmdVariables` to `{}`, so the if statement below wouldn't get entered and the quickpick for sqlcmd variables wasn't getting shown even when a project did have sqlcmd variables. 

This caused problems after https://github.com/microsoft/vscode-mssql/pull/17683 was merged, because the DacFxService was expecting a map, but was getting `{}`